src/rocdecode/vaapi/vaapi_videodecoder.cpp

```diff
@@ -301,6 +304,30 @@ rocDecStatus VaapiVideoDecoder::SubmitDecode(RocdecPicParams *pPicParams) {
             break;
         }
 
+        case rocDecVideoCodec_AV1: {
+            pPicParams->pic_params.av1.current_frame = curr_surface_id;
+            for (int i = 0; i < 8; i++) {
+                if (pPicParams->pic_params.av1.ref_frame_idx[i] != 0xFF) {
+                    if (pPicParams->pic_params.av1.ref_frame_idx[i] >= va_surface_ids_.size() || pPicParams->pic_params.av1.ref_frame_idx[i] < 0) {
+                        ERR("Reference frame index exceeded the VAAPI surface pool limit.");
+                        return ROCDEC_INVALID_PARAMETER;
+                    }
+                    pPicParams->pic_params.av1.ref_frame_idx[i] = va_surface_ids_[pPicParams->pic_params.av1.ref_frame_idx[i]];
```

Comment:
We should assign VA surface IDs to ref_frame_map[8], not ref_frame_idx[7]. ref_frame_idx[7] is the indices to ref_frame_map[].